src/plugins/httpUpload/HttpUploadPlugin.ts

```diff
@@ -254,6 +255,32 @@ export default class HttpUploadPlugin extends AbstractPlugin {
             }
          }
       }
+      else
+          if (body.length===1 && !message.isEncrypted())
+          {
+              var link = body.text();
+
+              var pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
+                '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
+                '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
+                '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
+                '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
+                '(\\#[-a-z\\d_]*)?$','i'); // fragment locator
+              if (!!pattern.test(link)&&link.lastIndexOf('.')>0)
+              {
+                   let suffix = link.substring(link.lastIndexOf('.')+1).toLowerCase();
+
+                   if (link.endsWith('.jpeg')||link.endsWith('.jpg')||link.endsWith('.png')||link.endsWith('.gif')||link.endsWith('.svg'))
```

Comment:
could also be written as `['jpeg', ...].indexOf(suffix) > -1`.

Reply:
changed...